### PR TITLE
Refactor the classes and move the templates under template, REST serv…

### PR DESCRIPTION
**Pull request by cmoulliard**
- Refactor the classes and move the:
  - HTML pages within the package template
  - REST Panache services to the package service
- Issue: #20

### CODE REVIEW

File: servicebox-app/src/main/java/io/halkyon/service/ClaimResource.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package io.halkyon.service;
+
+import io.halkyon.model.Claim;
+import io.quarkus.hibernate.orm.rest.data.panache.PanacheEntityResource;
+import io.quarkus.rest.data.panache.ResourceProperties;
+
+
+/***
+ * The "@ResourceProperties" annotation of the Quarkus JAX-RS panache extension generates the JAX-RS CRUD endpoints for the ClaimResource interface.
```
Comment (aureamunoz) [inline servicebox-app/src/main/java/io/halkyon/service/ClaimResource.java:9]: `@ResourceProperties` annotation that can be used to customize certain features of the resource (here the path) but is not responsible for generating the Resource implementation.
REST Data with Panache generates JAX-RS resources based on the presence of the interface which extends the PanacheEntityResource<Claim, Long>, being `Claim`the entity.

File: servicebox-app/src/main/java/io/halkyon/template/ClaimPage.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
 @Path("/claims")
 @Produces(MediaType.APPLICATION_JSON)
 @Consumes(MediaType.APPLICATION_JSON)
-public class ClaimExtendedResource {
+public class ClaimPage {
```
Comment (aureamunoz) [inline servicebox-app/src/main/java/io/halkyon/template/ClaimPage.java:18]: Rename to ClaimPageResource?

File: servicebox-app/src/main/resources/templates/ClaimPage/list.html [hunk 3]
```diff
@@ -36,14 +36,14 @@ <h1>Claims</h1>
                 <tbody id="claim-list">
                   {#for claim in claims}
                       <!-- Include the HTML template rendering the claim item -->
-                      {#include ClaimExtendedResource/item.html claim=claim /}
+                      {#include ClaimPage/item.html claim=claim /}
```
Comment (aureamunoz) [inline servicebox-app/src/main/resources/templates/ClaimPage/list.html:39]: Rename according to the resource renaming

**Review by aureamunoz — CHANGES_REQUESTED** [review]

File: servicebox-app/src/main/java/io/halkyon/service/ClaimResource.java (hunk 1 above)
Comment (cmoulliard) [reply servicebox-app/src/main/java/io/halkyon/service/ClaimResource.java:9]: do you then suggest to review the wording of the javadoc ?

File: servicebox-app/src/main/java/io/halkyon/template/ClaimPage.java (hunk 2 above)
Comment (cmoulliard) [reply servicebox-app/src/main/java/io/halkyon/template/ClaimPage.java:18]: OK

File: servicebox-app/src/main/resources/templates/ClaimPage/list.html (hunk 3 above)
Comment (cmoulliard) [reply servicebox-app/src/main/resources/templates/ClaimPage/list.html:39]: OK. Let's do that

File: servicebox-app/src/main/java/io/halkyon/service/ClaimResource.java (hunk 1 above)
Comment (aureamunoz) [reply servicebox-app/src/main/java/io/halkyon/service/ClaimResource.java:9]:
```suggestion
/***
 * REST Data with Panache extension generates JAX-RS resources based on the presence of this interface.
 * The @ResourceProperties annotation is used to customize the path of the resource.
 * More information is available within the quarkus documentation: <a href="https://quarkus.io/guides/rest-data-panache">https://quarkus.io/guides/rest-data-panache</a>
 * The endpoints generated support as HTTP request/response, the `application/json` content type.
 * The endpoints can be overridden, extended using the @see io.halkyon.template.ClaimPage class.
 */
```

**Review by aureamunoz — CHANGES_REQUESTED** [review]

**Review by cmoulliard — COMMENTED** [review]
LGTM

**Pull request merged**